src/StackExchange.Redis/PhysicalConnection.cs

```diff
@@ -1551,6 +1551,27 @@ private void MatchResult(in RawResult result)
             }
             _readStatus = ReadStatus.MatchResultComplete;
             _activeMessage = null;
+
+            static bool TryGetPubSubPayload(in RawResult value, out RedisValue parsed, bool allowArraySingleton = true)
+            {
+                if (value.IsNull)
+                {
+                    parsed = RedisValue.Null;
+                    return true;
+                }
+                switch (value.Type)
+                {
+                    case ResultType.Integer:
+                    case ResultType.SimpleString:
+                    case ResultType.BulkString:
+                        parsed = value.AsRedisValue();
+                        return true;
+                    case ResultType.MultiBulk when value.ItemsCount == 1:
+                        return TryGetPubSubPayload(in value[0], out parsed, allowArraySingleton: false);
```

Comment:
FWIW I think this would be simpler to take the first element as a RedisValue instead of recursion just for ease of understanding/maintenance, but there may be a future case of nested arrays I'm unaware of.

Reply:
@NickCraver yah, avoiding edge cases and avoiding having to repeat the "what combinations are allowed" (the `switch`)